pkg/gits/credentialhelper/git_credentialhelper.go

```diff
@@ -0,0 +1,111 @@
+package credentialhelper
+
+import (
+	"bufio"
+	"fmt"
+	"io"
+
+	"github.com/pkg/errors"
+)
+
+// GitCredentialsHelper is used to implement the git credential helper algorithm.
+// See also https://git-scm.com/docs/git-credential.
+type GitCredentialsHelper struct {
+	in               io.Reader
+	out              io.Writer
+	knownCredentials []GitCredential
+}
+
+// CreateGitCredentialsHelper creates an instance of a git credential helper. It needs to get passed the handles to read
+// the git credential data as well as write the response to. It also gets the list og known credentials.
+func CreateGitCredentialsHelper(in io.Reader, out io.Writer, credentials []GitCredential) (*GitCredentialsHelper, error) {
+	if in == nil {
+		return nil, errors.New("in parameter cannot be nil")
+	}
+
+	if out == nil {
+		return nil, errors.New("out parameter cannot be nil")
+	}
+
+	if credentials == nil {
+		return nil, errors.New("credentials parameter cannot be nil")
+	}
+
+	return &GitCredentialsHelper{
+		in:               in,
+		out:              out,
+		knownCredentials: credentials,
+	}, nil
+}
+
+// Run executes the specified git credential helper operation which must be one of get, store or erase.
+// NOTE: Currently only get is implemented.
+func (h *GitCredentialsHelper) Run(op string) error {
+	var err error
+
+	switch op {
+	case "get":
+		err = h.Get()
+	case "store":
+		// not yet implemented (HF)
+		fmt.Println("")
```

Comment:
Shouldn't we return an error here when is not implemented?

Reply:
I think in this case returning the empty string is the right thing. There is no need for an error. Git will send us the store request and we just ignore it.

Eventually, I think we could use this to store the user's credentials within the cluster/vault.